ethereum-consensus/src/state_transition/error.rs

```diff
@@ -189,6 +191,18 @@ pub enum InvalidWithdrawals {
     IncorrectWithdrawals { provided: Vec<Withdrawal>, expected: Vec<Withdrawal> },
 }
 
+#[derive(Debug, Error)]
+pub enum InvalidBlsToExecutionChange {
+    #[error("invalid bls signature for execution change {0:?}")]
+    ExecutionChange(BlsSignature),
+    #[error("validator index is out of bounds {0}")]
+    BadValidatorIndex(usize),
+    #[error("invalid withdrawal credentials prefix{0}")]
+    WithdrawalCredentialsPrefix(u8),
+    #[error("invalid withdrawal credentials public key{0:?}")]
+    WithdrawalCredentialsPublicKey(BlsPublicKey),
```

Comment:
this is a bit better name imo
```suggestion
    PublicKeyMismatch(BlsPublicKey),
```